### PR TITLE
Fluent-like updaters for Mongo 

### PR DESCRIPTION
**Pull request by ilyahryapko**

### CODE REVIEW

**Comment by ilyahryapko** [comment]
Removed Increment validation. Reason: Mongo throws an exception when increment value is not valid
Replaced dictionary with list. Reason: not obvious behavior with multiple same field updates. Mongo throws an exception when update operation affects same field twice or more.

**Comment by goremykin** [comment]
Add @SergeyLukashov or @yarmolovichalex please

File: src/app/Common/DAL/FluentUpdater/Updater.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Common.DAL.Documents;
+
+namespace Common.DAL.FluentUpdater
+{
+    public static class Updater<TDocument> where TDocument : BaseDocument
```
Comment (SergeyLukashov) [inline src/app/Common/DAL/FluentUpdater/Updater.cs:8]: @ilyahryapko Do not get the problem this class is solving. Does it simply wrap `MongoFluentUpdater` in order to mitigate usage of `new` operator in the repository calls?

```
await _userRepository.UpdateOneAsync(id, Updater<User>
   .Set(u => u.PasswordHash, newPassword.GetHash())
   .Set(u => u.ResetPasswordToken, string.Empty));
```
vs
```
await _userRepository.UpdateOneAsync(id, **new** MongoFluentUpdater<User>()
   .Set(u => u.PasswordHash, newPassword.GetHash())
   .Set(u => u.ResetPasswordToken, string.Empty));
```

File: src/app/Common/DAL/UpdateDocumentOperators/IncrementOperator.cs [hunk 2]
```diff
@@ -14,16 +14,6 @@ public class IncrementOperator<TDocument, TItem>: IUpdateOperator<TDocument> whe
         
         public IncrementOperator(Expression<Func<TDocument, TItem>> field, TItem incrementByValue)
         {
-            if (!IncrementOperatorValidator.IsValidFieldType(typeof(TItem)))
```
Comment (SergeyLukashov) [inline src/app/Common/DAL/UpdateDocumentOperators/IncrementOperator.cs:17]: 😜

**Comment by ilyahryapko** [comment]
@goremykin @SergeyLukashov 
Added "Apply" method for cases when entities could have common updates for some fields

File: src/app/Common/DAL/FluentUpdater/Updater.cs (hunk 1 above)
Comment (ilyahryapko) [reply src/app/Common/DAL/FluentUpdater/Updater.cs:9]: Yep. Simply because of that. "Updater" looks cleaner then new object with long name

**Pull request merged**